packages/scripts/src/admin-proposals/change-final-fee/0_Propose.ts

```diff
@@ -76,64 +74,93 @@ async function main() {
   const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
 
   const proposer = await await getContractInstance<ProposerV2Ethers>("ProposerV2");
-  const store = await getContractInstance<StoreEthers>("Store");
 
   // case mainnet
-  for (const token in tokensToUpdateFee) {
+  for (const token in tokensToUpdate) {
     const provider = getRetryProvider(1) as Provider;
 
-    const isWeth = token == "WETH";
-    const newFinalFee = isWeth ? newFinalFeeWeth : newFinalFeeUSD;
+    const tokenAddress = tokensToUpdate[token].mainnet;
+    const newFinalFee = tokensToUpdate[token].finalFee;
+    if (!tokenAddress) continue;
 
-    const tokenAddress = tokensToUpdateFee[token as keyof typeof tokensToUpdateFee].mainnet;
     const erc20 = await getContractInstanceWithProvider<ERC20Ethers>("ERC20", provider, tokenAddress);
     const decimals = await erc20.decimals();
 
+    // Check if the token is already whitelisted
+    const addressWhitelist = await getConnectedAddressWhitelist(1);
+    const isWhitelisted = await addressWhitelist.isOnWhitelist(tokenAddress);
+    if (!isWhitelisted) {
+      console.log(`Adding ${token} to the whitelist on mainnet...`);
+      const addAddressTx = await addressWhitelist.populateTransaction.addToWhitelist(tokenAddress);
+      if (!addAddressTx.data) throw "addAddressTx.data is null";
```

Comment:
good practice to use errors `throw new Error(...)`  so that you have a stack trace, or use assert which will create the error for you, but the logic is not intuitive since its inverted: `assert(addAddressTx.data,"addAddressTx.data is null")`.  not huge deal in a script like this though